account-kit/infra/src/alchemyTransport.ts

```diff
@@ -0,0 +1,213 @@
+import {
+  ChainNotFoundError,
+  ConnectionConfigSchema,
+  split,
+  type ConnectionConfig,
+  type NoUndefined,
+} from "@aa-sdk/core";
+import {
+  createTransport,
+  http,
+  type HttpTransportConfig,
+  type Transport,
+  type TransportConfig,
+} from "viem";
+import { AlchemyChainSchema } from "./schema.js";
+import { VERSION } from "./version.js";
+
+type EthereumProvider = { request(...args: any): Promise<any> };
+
+type Never<T> = T extends object
+  ? {
+      [K in keyof T]?: never;
+    }
+  : never;
+
+type SplitTransportConfig = {
+  alchemyConnection: ConnectionConfig;
+  nodeRpcUrl: string;
```

Comment:
I'm wondering if this should actually be called `nodeTransport` and take in a viem `Transport` type instead. That would allow devs more freedom, but the instantiation looks wilder.

Reply:
I'm a fan of passing the transport to escape hatch ourselves from upstream requests. Can we wrap a helper function to create a transport with the nodeRPCUrl? Is that much different than just the normal `http(...)`?

Reply:
yea it's basically just `http(...)` in most cases and I think it's a good escape hatch, but it does make the declaration slightly more verbose. I might later add a union type to this and turn it into `{alchemyConnection, nodeTransport}` if it becomes an issue